src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java

```diff
@@ -36,39 +38,46 @@ public SchemaPosition(ConnectionPool pool, Long serverID) {
 
 	public void start() {
 		this.thread = new Thread(this, "Position Flush Thread");
-		this.run.set(true);
 		thread.start();
 	}
 
-	public void stop() {
-		this.run.set(false);
-
+	public void stopLoop() throws TimeoutException {
+		this.requestStop();
 		thread.interrupt();
+		super.stopLoop();
```

Comment:
here, because I need to send `thread.interrupt()`, I had to override `stopLoop`.